### PR TITLE
feat: load page

### PR DESCRIPTION
**Pull request by petia-latinova**

### CODE REVIEW

File: batch-nft-reveal/app/components/load/Load.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+export function validateAddress(value): JSX.Element {
+  let error
+  if (!value) {
+    error = 'Required'
+  } else if (!/^0x[a-fA-F0-9]{40}$/.test(value)) {
+    error = 'Invalid address'
+  }
+  return error
+}
```
Comment (imollov) [inline batch-nft-reveal/app/components/load/Load.tsx:20]:
```suggestion
}

```

File: batch-nft-reveal/app/components/load/Load.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import {
+  Button,
+  Container,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+} from '@chakra-ui/react'
+
+export function validateAddress(value): JSX.Element {
```
Comment (imollov) [inline batch-nft-reveal/app/components/load/Load.tsx:12]:
```suggestion
export function validateAddress(value): string {
```

Comment (imollov) [reply batch-nft-reveal/app/components/load/Load.tsx:12]: Why is this function exported from the component?

File: batch-nft-reveal/app/components/load/Load.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  let error
+  if (!value) {
+    error = 'Required'
+  } else if (!/^0x[a-fA-F0-9]{40}$/.test(value)) {
+    error = 'Invalid address'
+  }
+  return error
+}
+export const Load = () => (
```
Comment (imollov) [inline batch-nft-reveal/app/components/load/Load.tsx:21]:
```suggestion
export const Load = (): JSX.Element => (
```

File: batch-nft-reveal/app/components/load/Load.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    initialValues={{
+      address: '',
+    }}
+    onSubmit={(values) => {
+      console.log(values)
+    }}
+  >
+    {({ errors, touched, isValidating }) => (
+      <Form>
+        <Container maxW="mt" centerContent>
```
Comment (imollov) [inline batch-nft-reveal/app/components/load/Load.tsx:32]: Content shouldn't be centered. See the fullstack app for reference.

File: batch-nft-reveal/app/components/load/index.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export { Load } from './Load'
```
Comment (imollov) [inline batch-nft-reveal/app/components/load/index.ts:1]:
```suggestion
export { Load } from './Load'

```

File: batch-nft-reveal/app/components/load/Load.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+          <FormControl mt="4" isInvalid={!!errors.address}>
+            <FormLabel htmlFor="url">Address</FormLabel>
```
Comment (imollov) [inline batch-nft-reveal/app/components/load/Load.tsx:34]: url?

File: batch-nft-reveal/app/pages/load.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import { Heading } from '@chakra-ui/react'
+import { Load } from '../components/load'
+import { Section, Layout } from '../components/layout'
+
+function LoadPage(): JSX.Element {
+
```
Comment (imollov) [inline batch-nft-reveal/app/pages/load.tsx:7]:
```suggestion
```

File: batch-nft-reveal/app/components/load/Load.tsx (hunk 2 above)
Comment (petia-latinova) [reply batch-nft-reveal/app/components/load/Load.tsx]: It's a mistake. I changed it.

File: batch-nft-reveal/app/components/load/Load.tsx (hunk 6 above)
Comment (petia-latinova) [reply batch-nft-reveal/app/components/load/Load.tsx]: should be address

**Review by ivaylonikolov7 — APPROVED** [review]
LGTM, maybe you could make it redirect to /collection/{some address} so when my PR is ready they can work together?

File: batch-nft-reveal/app/components/load/Load.tsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+      error = 'Required'
+    } else if (!/^0x[a-fA-F0-9]{40}$/.test(value)) {
+      error = 'Invalid address'
+    }
+    return error
+  }
+
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <Container maxW="mt">
```
[the remark this answers is not in the document]
Comment (petia-latinova) [reply batch-nft-reveal/app/components/load/Load.tsx]: You are right.it is unnecessary to add it there. I removed it.

**Review by ivaylonikolov7 — APPROVED** [review]

**Comment by imollov** [comment]
@petia-latinova Submit button is not enabled when there's a valid address:

Screenshot 2022-08-22 at 15 39 08

**Comment by petia-latinova** [comment]
> @petia-latinova Submit button is not enabled when there's a valid address:
> 
> Screenshot 2022-08-22 at 15 39 08

Probably you didn't connected to metamask.

**Comment by imollov** [comment]
> > @petia-latinova Submit button is not enabled when there's a valid address:
> > Screenshot 2022-08-22 at 15 39 08
> 
> Probably you didn't connected to metamask.

Collection page is going to work in read-only mode.

**Review by imollov — APPROVED** [review]

**Pull request merged**